ccdproc/image_collection.py

```diff
@@ -69,6 +69,7 @@ def __init__(self, location=None, keywords=None, info_file=None,
         self._location = location
         self._filenames = filenames
         self._files = []
+        self._info_file = info_file
```

Comment:
New attributes are always tricky because all sort of modifications on the instance need to keep the class invariants. Each added invariant makes the class more difficult. For example can you garantuee that any change on an instance correctly handles this attribute?

I think that may be much more trouble than it's beneficial.